jupytext/jupytext_config.py

```diff
@@ -0,0 +1,64 @@
+from argparse import ArgumentParser
+
+from .labconfig import LabConfig
+
+# list of subcommands - filled by SubCommand.__init__
+SUBCOMMANDS = []
+
+
+class SubCommand:
+    def __init__(self, name, help):
+        self.name = name
+        self.help = help
+        SUBCOMMANDS.append(self)
+
+    def main(self, args):
+        """
+        return 0 if all goes well
+        """
+        print(
+            f"{self.__class__.__name__}: redefine main() to implement this subcommand"
+        )
+        return 1
+
+
+class List(SubCommand):
+    def __init__(self):
+        super().__init__("list", "Display current settings in labconfig/")
+
+    def main(self, args):
+        LabConfig().read().list()
+        return 0
+
+    def fill_parser(self, subparser):
+        pass
+
+
+class SetDefaultViewer(SubCommand):
+    def __init__(self):
+        super().__init__("set-default-viewer", "Set default viewers for JupyterLab")
+
+    def main(self, args):
+        LabConfig().read().set_default_viewers(args.language).write()
+        return 0
+
+    def fill_parser(self, subparser):
+        subparser.add_argument(
+            "language", nargs="*", help="the language(s) that the viewer applies to"
+        )
+
+
+# create the subcommands
```

Comment:
Maybe `SUBCOMMANDS=[List(), SetDefaultViewer()]` would be more explicit?

Reply:
see af22579f6d94b373b96efe4e5485a21c80d2e941